### PR TITLE
Added unit testing

### PR DESCRIPTION
**Pull request by Arch-Noize**
With the help of RSpec, we've added unit tests for our classes and their methods. You can find tests for:
- `Book`
- `Person`
- `Student`
- `Teacher`
- `Classroom`
- `Rental`
- and all classes that we created while implementing the Decorator design pattern

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+require 'rspec'
+require './classes/book'
+require './classes/person_class'
+
+describe Book do
+  context 'Given a Book named The Sphere by Michael Crichton' do
+    before :each do
```
Comment (V-Blaze) [inline spec/book_spec.rb:8]: - You have done great so far :+1: But there are a few requirements you need to meet before moving to the next milestone. :smile: :wave: 

- - Screenshot reference:
Selection_035

- [ ] According to the project requirement, you are meant to have ` unit tests for all the methods in all the classes`. Currently in your `book_spec`, you do not have a test case for the `add_rental` and `to_json` methods in the book class. i suggest you write the tests to cover these cases. :smile: :waffle:  :+1:

File: spec/student_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+require 'rspec'
+require './classes/student_class'
+require './classes/classroom'
+
+describe Student do
+  context 'Given a Student named Santiago' do
+    before :each do
+      @classroom = Classroom.new('Gepgraphy')
+      @student = Student.new(age: 24, name: 'Santiago', parent_permission: false, classroom: @classroom)
+    end
```
Comment (V-Blaze) [inline spec/student_spec.rb:10]: - [ ] Your `student_spec` does not have a test for the `play_hooky` method in the student class. I suggest you add this one too. :smile: :small_airplane: :+1:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @Arch-Noize, @Megagig  

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- All current tests are passing✔
-  All test files placed in a different directory👋 
-  Clean and well written tests ✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ibizugbe — APPROVED** [review]
Hi @Arch-Noize @Megagig , 

approved


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights✅✅✅✅

- [x] The add_rental and  play_hocky methods are tested as requested by the previous reveiwer👍🏼.
- [x] All tests are passing👍🏼.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**